File: bro-agent-api/src/test/java/com/ericsson/adp/mgmt/bro/api/test/TestAgentBehavior.java
/*------------------------------------------------------------------------------
 *******************************************************************************
 * COPYRIGHT Ericsson 2019
 *
 * The copyright to the computer program(s) herein is the property of
 * Ericsson Inc. The programs may be used and/or copied only with written
 * permission from Ericsson Inc. or in accordance with the terms and
 * conditions stipulated in the agreement/contract under which the
 * program(s) have been supplied.
 *******************************************************************************
 *----------------------------------------------------------------------------*/
package com.ericsson.adp.mgmt.bro.api.test;

import com.ericsson.adp.mgmt.bro.api.agent.AgentBehavior;
import com.ericsson.adp.mgmt.bro.api.agent.BackupExecutionActions;
import com.ericsson.adp.mgmt.bro.api.agent.CancelActions;
import com.ericsson.adp.mgmt.bro.api.agent.PostRestoreActions;
import com.ericsson.adp.mgmt.bro.api.agent.RestoreExecutionActions;
import com.ericsson.adp.mgmt.bro.api.agent.RestorePreparationActions;
import com.ericsson.adp.mgmt.bro.api.agent.BackupPreparationActions;
import com.ericsson.adp.mgmt.bro.api.agent.PostBackupActions;
import com.ericsson.adp.mgmt.bro.api.registration.RegistrationInformation;

public class TestAgentBehavior implements AgentBehavior {

    private boolean executedBackup;
    private boolean preparedBackup;
    private boolean performedPostBackup;
    private boolean executedRestore;
    private boolean preparedForRestore;
    private boolean performedPostRestore;
    private boolean cancelledAction;

    @Override
    public RegistrationInformation getRegistrationInformation() {
        return RegistrationInformationUtil.getTestRegistrationInformation();
    }

    @Override
    public void prepareForBackup(BackupPreparationActions backupPreparationActions) {
        preparedBackup = true;
        backupPreparationActions.sendStageComplete(true, "Test Prepare backup.");
    }

    @Override
    public void postBackup(PostBackupActions postBackupActions) {
        performedPostBackup = true;
        postBackupActions.sendStageComplete(true, "Test Post backup.");
    }

    @Override
    public void executeBackup(final BackupExecutionActions backupExecutionActions) {
        this.executedBackup = true;
        backupExecutionActions.backupComplete(true, "Test backup");
    }

    @Override
    public void prepareForRestore(final RestorePreparationActions restorePreparationActions) {
        this.preparedForRestore = true;
        restorePreparationActions.sendStageComplete(true, "stage success");
    }

    @Override
    public void postRestore(final PostRestoreActions postRestoreActions) {
        this.performedPostRestore = true;
        postRestoreActions.sendStageComplete(true, "stage success");
    }

    @Override
    public void executeRestore(final RestoreExecutionActions restoreExecutionActions) {
        this.executedRestore = true;
        restoreExecutionActions.restoreComplete(true, "stage success");
    }

    @Override
    public void cancelAction(final CancelActions cancelActions) {
        this.cancelledAction = true;
        cancelActions.sendStageComplete(true, "stage success");
    }

    public boolean executedBackup() {
        return this.executedBackup;
    }

    public boolean executedRestore() {
        return this.executedRestore;
    }

    public boolean preparedForRestore() {
        return this.preparedForRestore;
    }

    public boolean performedPostRestore() {
        return this.performedPostRestore;
    }

    public boolean cancelledAction() {
        return this.cancelledAction;
    }

    public boolean hasPreparedBackup() {
        return preparedBackup;
    }

    public boolean hasPerformedPostBackup() {
        return performedPostBackup;
    }

    public void reset() {
        this.executedBackup = false;
        this.executedRestore = false;
        this.preparedBackup = false;
        this.preparedForRestore = false;
        this.performedPostBackup = false;
        this.performedPostRestore = false;
        this.cancelledAction = false;
    }

}

File: bro-agent-api/src/test/python/TestAgentBehavior.py
from bro_agent.agent.AgentBehavior import AgentBehavior
from bro_agent.agent.BackupExecutionActions import BackupExecutionActions
from bro_agent.agent.CancelActions import CancelActions
from bro_agent.agent.PostRestoreActions import PostRestoreActions
from bro_agent.agent.RestoreExecutionActions import RestoreExecutionActions
from bro_agent.agent.RestorePreparationActions import RestorePreparationActions
from bro_agent.registration.RegistrationInformation import RegistrationInformation
from RegistrationInformationUtil import RegistrationInformationUtil

class TestAgentBehavior(AgentBehavior):
    def __init__(self):
        self.executed_backup = False
        self.executed_restore = False
        self.prepared_for_restore = False
        self.performed_post_restore = False
        self.cancelled_action = False

    def get_registration_information(self):
        return RegistrationInformationUtil.get_test_registration_information()

    def execute_backup(self, backupExecutionActions):
        self.executed_backup = True
        backupExecutionActions.backup_complete(True, "Test backup")

    def prepare_for_restore(self, restorePreparationActions):
        self.prepared_for_restore = True
        restorePreparationActions.send_stage_complete(True, "stage success")

    def post_restore(self, postRestoreActions):
        self.performed_post_restore = True
        postRestoreActions.send_stage_complete(True, "stage success")

    def execute_restore(self, restoreExecutionActions):
        self.executed_restore = True
        restoreExecutionActions.restore_complete(True, "stage success")

    def cancel_action(self, cancelActions):
        self.cancelled_action = True
        cancelActions.send_stage_complete(True, "stage success")

    def executedBackup(self):
        return self.executed_backup

    def executedRestore(self):
        return self.executed_restore

    def preparedForRestore(self):
        return self.prepared_for_restore

    def performedPostRestore(self):
        return self.performed_post_restore

    def cancelledAction(self):
        return self.cancelled_action

    def reset(self):
        self.executed_backup = False
        self.executed_restore = False
        self.prepared_for_restore = False
        self.performed_post_restore = False
        self.cancelled_action = False
